unstructured/file_utils/filetype.py

```diff
@@ -105,102 +105,183 @@ def __lt__(self, other):
     ".rtf": FileType.RTF,
 }
 
-
-def detect_filetype(
+def detect_filetype(    
     filename: Optional[str] = None, file: Optional[IO] = None
 ) -> Optional[FileType]:
-    """Use libmagic to determine a file's type. Helps determine which partition brick
-    to use for a given file. A return value of None indicates a non-supported file type."""
-    if filename and file:
-        raise ValueError("Only one of filename or file should be specified.")
-
-    if filename:
-        _, extension = os.path.splitext(filename)
-        extension = extension.lower()
-        mime_type = magic.from_file(filename, mime=True)
-    elif file is not None:
-        extension = None
-        # NOTE(robinson) - the python-magic docs recommend reading at least the first 2048 bytes
-        # Increased to 4096 because otherwise .xlsx files get detected as a zip file
-        # ref: https://github.com/ahupp/python-magic#usage
-        mime_type = magic.from_buffer(file.read(4096), mime=True)
```

Comment:
In this block, if `libmagic` is unavailable let's raise an `ImportError` that says something along the lines of `"libmagic is unavailable. Filetype detection on file-like objects requires libmagic. Please install libmagic and try again."`